### PR TITLE
feat(secrets): support scanning of secrets in hidden paths

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- created a new method to do the filtering, when the user decides to scan all files, because the logic in the old function didn't play nice with the `excluded_paths` parameter
- also excluded `.idea`, `.git` and `venv` from scanning

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/secrets/utils.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def filter_excluded_paths(
+    root_dir: str,
+    names: list[str] | list[os.DirEntry[str]],
+    excluded_paths: Iterable[str] | None,
+) -> None:
+    """Special build of checkov.common.runners.base_runner.filter_ignored_paths for Secrets scanning"""
+
+    # old logic
```
Comment (omryMen) [inline checkov/secrets/utils.py:20]: is it planned to keep the old logic ? or is it just temp ?

**Review by omryMen — APPROVED** [review]
one question..

File: checkov/secrets/utils.py (hunk 1 above)
Comment (gruebel) [reply checkov/secrets/utils.py:20]: it will stay, because this for the case a user uses the `--skip-path` flag. Should I maybe reword it? I commented it with old, because it is from the old function 😄

File: checkov/secrets/runner.py [hunk 2]
```diff
@@ -173,8 +172,14 @@ def run(
                     block_list_secret_scan = runner_filter.block_list_secret_scan or []
                     block_list_secret_scan_lower = [file_type.lower() for file_type in block_list_secret_scan]
                     for root, d_names, f_names in os.walk(root_folder):
-                        filter_ignored_paths(root, d_names, excluded_paths)
-                        filter_ignored_paths(root, f_names, excluded_paths)
+                        if enable_secret_scan_all_files:
+                            # 'excluded_paths' shouldn't include the static paths from 'EXCLUDED_PATHS'
+                            # they are separately referenced inside the 'filter_excluded_paths' function
+                            filter_excluded_paths(root_dir=root, names=d_names, excluded_paths=runner_filter.excluded_paths)
+                            filter_excluded_paths(root_dir=root, names=f_names, excluded_paths=runner_filter.excluded_paths)
```
Comment (pazbechor) [inline checkov/secrets/runner.py:179]:
```suggestion
                            filter_excluded_paths(root_dir=root, names=[*d_names, *f_names], excluded_paths=runner_filter.excluded_paths)
```

File: checkov/secrets/runner.py [hunk 3]
```diff
@@ -173,8 +172,14 @@ def run(
...
+                        else:
+                            filter_ignored_paths(root, d_names, excluded_paths)
+                            filter_ignored_paths(root, f_names, excluded_paths)
```
[the remark this answers is not in the document]
Comment (gruebel) [reply checkov/secrets/runner.py:182]: the function adjusts the content of `names` parameter, that's why I need the original `list`, don't want to adjust the actual logic 😅

**Review by pazbechor — APPROVED** [review]
